File: Day 25 : Running Time and Complexity/Solution.js
function processData(input) {
        var arr = input.split("\n");
        for(let i = 1; i <= parseInt(arr[0]); i++){
            let ele = parseInt(arr[i]);
            if(ele <= 1) {
                console.log("Not prime");
            } else {
                let divisors = 1;
                for(let j = 2; j <= Math.sqrt(ele); j++) {
                    if(ele % j == 0){
                        console.log("Not prime");
                        divisors++;
                        break;
                    }
                }
                if(divisors == 1)
                    console.log("Prime");
            }
        }
    } 
    
processData(input.readline());

File: Day 25 : Running Time and Complexity/Solution.rb
T = gets.to_i

T.times do
  ele = gets.to_i
  t = Math.sqrt(ele).round(0)

  if ele <= 1
    puts "Not prime"
  else
    d = 1
    for j in 2..t
      if ele % j == 0
        puts "Not prime"
        d += 1
        break
      end
    end
    if d == 1
      puts "Prime"
    end
  end
end
